pkg/Cpanel/Security/Advisor/Assessors/Permissions.pm

```diff
@@ -55,7 +55,11 @@ sub _check_for_unsafe_permissions {
             }
         }
         if ( !$perms_ok ) {
-            my $expected_mode = sprintf( "%04o", $expected_attributes->{'perms'} );
+            my $expected_mode = '';
+            foreach my $perms ( @{ $expected_attributes->{'perms'} } ) {
+                $expected_mode .= sprintf( "%04o ", $perms );
```

Comment:
Couldn't we use a map and join here, like this?
```perl
my $expected_mode = join(' ', map { sprintf('%04o', $_) } @{$expected_attributes->{'perms'}});
```
That would avoid the need for the cleanup on the next line and would be marginally more efficient.